### PR TITLE
[New Profile] Fix TypeError on candidate creation

### LINKED ISSUE #6224: [new_profile] create a new profile Argument 1 passed must be of the type array, string given
**Describe the bug**
Trying to create a new profile is throwing this error:

```
[Wed Apr 01 17:47:44.142984 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP TypeError:  Argument 1 passed to LORIS\\Http\\Response\\JSON\\Created::__construct() must be of the type array, string given, called in /Users/alizee/Development/GitHub/McGill/Loris/modules/new_profile/php/new_profile.class.inc on line 111 in /Users/alizee/Development/GitHub/McGill/Loris/src/Http/Response/JSON/Created.php on line 42, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143186 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP Stack trace:, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143194 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   1. {main}() /Users/alizee/Development/GitHub/McGill/Loris/htdocs/index.php:0, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143239 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   2. LORIS\\Middleware\\ContentLength->process() /Users/alizee/Development/GitHub/McGill/Loris/htdocs/index.php:47, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143246 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   3. LORIS\\Middleware\\ResponseGenerator->process() /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/ContentLength.php:52, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143252 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   4. LORIS\\Router\\BaseRouter->handle() /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/ResponseGenerator.php:50, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143258 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   5. LORIS\\Router\\ModuleRouter->handle() /Users/alizee/Development/GitHub/McGill/Loris/src/Router/BaseRouter.php:110, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143263 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   6. LORIS\\Middleware\\AuthMiddleware->process() /Users/alizee/Development/GitHub/McGill/Loris/src/Router/ModuleRouter.php:75, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143268 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   7. LORIS\\Middleware\\ResponseGenerator->process() /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/AuthMiddleware.php:63, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143273 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   8. LORIS\\new_profile\\Module->handle() /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/ResponseGenerator.php:50, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143279 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP   9. LORIS\\new_profile\\New_Profile->process() /Users/alizee/Development/GitHub/McGill/Loris/php/libraries/Module.class.inc:412, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143286 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP  10. LORIS\\Middleware\\PageDecorationMiddleware->process() /Users/alizee/Development/GitHub/McGill/Loris/php/libraries/NDB_Page.class.inc:696, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143292 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP  11. LORIS\\Middleware\\UserPageDecorationMiddleware->process() /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/PageDecorationMiddleware.php:56, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143326 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP  12. LORIS\\new_profile\\New_Profile->handle() /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/UserPageDecorationMiddleware.php:233, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143336 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP  13. LORIS\\Http\\Response\\JSON\\Created->__construct() /Users/alizee/Development/GitHub/McGill/Loris/modules/new_profile/php/new_profile.class.inc:111, referer: http://localhost/new_profile/
[Wed Apr 01 17:47:44.143624 2020] [php7:notice] [pid 34639] [client ::1:53274] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to LORIS\\Http\\Response\\JSON\\Created::__construct() must be of the type array, string given, called in /Users/alizee/Development/GitHub/McGill/Loris/modules/new_profile/php/new_profile.class.inc on line 111 and defined in /Users/alizee/Development/GitHub/McGill/Loris/src/Http/Response/JSON/Created.php:42\nStack trace:\n#0 /Users/alizee/Development/GitHub/McGill/Loris/modules/new_profile/php/new_profile.class.inc(111): LORIS\\Http\\Response\\JSON\\Created->__construct('{"candID":"5231...')\n#1 /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/UserPageDecorationMiddleware.php(233): LORIS\\new_profile\\New_Profile->handle(Object(Laminas\\Diactoros\\ServerRequest))\n#2 /Users/alizee/Development/GitHub/McGill/Loris/src/Middleware/PageDecorationMiddleware.php(56): LORIS\\Middleware\\UserPageDecorationMiddleware->process(Object(Laminas\\Diactoros\\ServerRequest), Object(LORIS\\new_profile\\New_Profile))\n#3 /Users/alizee/Development/GitHub/McGill/Loris/php/libraries/NDB_Page.class.inc(69 in /Users/alizee/Development/GitHub/McGill/Loris/src/Http/Response/JSON/Created.php on line 42, referer: http://localhost/new_profile/

```

**To Reproduce**
Steps to reproduce the behavior (attach screenshots if applicable):
1. Go to new_profile
2. Click to create new profile
3. See error in logs

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Updates the JSON Response to the new idiom which solves the type error.

#### Testing instructions (if applicable)

1. Create a new profile. It should succeed.

#### Link(s) to related issue(s)

* Resolves #6224

### CODE REVIEW

File: modules/new_profile/php/new_profile.class.inc [hunk 1]
```diff
@@ -114,28 +114,13 @@ class New_Profile extends \NDB_Form
...
             if (!empty($this->_error)) {
-                return (new \LORIS\Http\Response())
-                    ->withHeader("Content-Type", "application/json")
-                    ->withStatus(400)
-                    ->withBody(
-                        new \LORIS\Http\StringStream(
-                            json_encode($this->_error)
-                        )
-                    );
+                return new \LORIS\Http\Response($this->_error);
```
Comment (driusan) [inline modules/new_profile/php/new_profile.class.inc:117]: This doesn't change the type, but loses a lot of information (such as the HTTP response code and content type..)

File: modules/new_profile/php/new_profile.class.inc [hunk 2]
```diff
@@ -114,28 +114,13 @@ class New_Profile extends \NDB_Form
...
-            return (new \LORIS\Http\Response())
-                ->withHeader("Content-Type", "application/json")
-                ->withStatus(201)
-                ->withHeader("Allow", "POST")
-                ->withBody(
-                    new \LORIS\Http\StringStream(
-                        json_encode($result)
-                    )
-                );
+            return new \LORIS\Http\Response\JSON\Created($result);
```
Comment (driusan) [inline modules/new_profile/php/new_profile.class.inc:123]: This doesn't change the return type either other than using a sub-class of response. It also loses the Allow header.

File: modules/new_profile/php/new_profile.class.inc (hunk 1 above)
Comment (johnsaigle) [reply modules/new_profile/php/new_profile.class.inc:117]: You're right, this should be BadRequest.
```suggestion
                return new \LORIS\Http\Response\JSON\BadRequest($this->_error);
```

File: modules/new_profile/php/new_profile.class.inc (hunk 2 above)
Comment (johnsaigle) [inline modules/new_profile/php/new_profile.class.inc:123]:
```suggestion
            return new \LORIS\Http\Response\JSON\Created($result)
                ->withHeader("Allow", "POST");
```

I'll add the Allow header back.

These changes do fix the bug if you test manually.

**Comment by johnsaigle** [comment]
The previous function required an array argument and was being given a string argument. Instead of changing the argument, this PR changes the function to the newer/cleaner idiom.

**Pull request merged**